src/map/mod.rs

```diff
@@ -6,6 +6,22 @@ use bevy::{
     prelude::{Component, Entity, Handle, Image, Reflect},
 };
 
+/// For the convenience in implementing `Add<Vec2>` for Vec2-like structures which are common in
+/// this module.
+macro_rules! impl_add_vec2 {
+    ($ty:ty $(, $inner_ty:ty)?) => {
+        impl std::ops::Add<Vec2> for $ty {
+            type Output = Self;
+            fn add(self, rhs: Vec2) -> Self::Output {
+                Self {
+                    x: self.x + rhs.x $(as $inner_ty)?,
```

Comment:
I'm not sure I like the `as u32` here since it hides potential unwanted side effects in this macro that would be difficult for a user to debug.